src/pool.c

```diff
@@ -173,6 +173,8 @@ hsk_pool_init(hsk_pool_t *pool, const uv_loop_t *loop) {
   pool->pending_count = 0;
   pool->block_time = 0;
   pool->getheaders_time = 0;
+  pool->user_agent = (char *)malloc(255);
```

Comment:
This should be freed in `hsk_pool_uninit` but if it's fixed size maybe it shouldn't use malloc here

Reply:
I had trouble with this because its not a fixed size. There is a max size and the string starts out as `hsd:v0.0.1` but then more characters are added to that later. I think you're right about free though, will add and force push

Reply:
ok updated with `free`

Reply:
> I had trouble with this because its not a fixed size. There is a max size and the string starts out as hsd:v0.0.1

oh I see i meant something like `char user_agent[255]` defined in pool struct then something like

```C
strcpy(pool->user_agent, HSK_USER_AGENT);
strcat(pool->user_agent, "extra string ...");
```
but yeah this works too

Reply:
> oh I see i meant something like `char user_agent[255]` defined in pool struct then something like

this is one thing I tried but then I got `array type char can not be assigned` or something

Reply:
confirmed leak: 

```

leaks Report Version: 4.0
Process 59478: 171 nodes malloced for 77 KB
Process 59478: 1 leak for 256 total leaked bytes.

    1 (256 bytes) ROOT LEAK: 0x1091059b0 [256]  length: 12  "/hnsd:1.0.0/"
```

and confirmed no leak after adding `free()`